src/vault/pass.rs

```diff
@@ -0,0 +1,151 @@
+use core::future::Ready;
+use mnemonic::Language;
+use prs_lib::{
+    crypto::{self, IsContext, Proto},
+    store::{FindSecret, Store},
+    Plaintext,
+};
+
+use crate::{RootAccount, Vault};
+
+/// A vault that holds secrets in memory
+pub struct Pass {
+    store: Store,
+    root: Option<RootAccount>,
+    auto_generate: Option<Language>,
+}
+
+const SERVICE_NAME: &str = "libwallet_service/";
+
+impl Pass {
+    /// Create a new OSKeyring vault for the given user.
+    /// The optional `lang` instructs the vault to generarte a backup phrase
+    /// in the given language in case one does not exist.
+    pub fn new<P: AsRef<str>>(store_path: P, lang: impl Into<Option<Language>>) -> Self {
+        let store = Store::open(store_path).unwrap();
+
+        Pass {
+            store,
+            root: None,
+            auto_generate: lang.into(),
+        }
+    }
+
+    fn get_key(&self, credentials: &PassCreds) -> Result<RootAccount, Error> {
+        let mut secret_path = String::from(SERVICE_NAME);
+        secret_path.push_str(&credentials.secret_path);
+
+        let secret = match self.store.find(Some(secret_path)) {
+            FindSecret::Exact(secret) => Some(secret),
+            FindSecret::Many(secrets) => {
+                if secrets.is_empty() {
```

Comment:
Instead of `.get(0)` you can do `.first()` and since it returns an option that will be `None` if the Vec is empty there's no need to check. Also Options have the `.cloned()` method to convert the &T to T.

Reply:
Gotcha!